### ISSUE TITLE
Big part of system configuration lost at night

### ISSUE THREAD

**Issue by gallogabor** [opnsense/core#6051]
I read the contributing doc, and then searched for similar open or closed problems without success.

**Describe the bug**

TL;DR: A big chunk of the configuration disappeared last night unexpectedly.

Last night a colleague of mine added users to the FreeRADIUS users database and modified some other user's description.
After pressing the Apply button on FreeRADIUS plugin Users page, after one of the modifications, he lost connectivity to the box and cannot proceed adding/editing users.
Today morning they called me, that no internet access available on the workstations. I opened the web admin from home (it worked somehow) and saw that no gateways defined, no IP addess on the main (fixed public address) WAN interface, other two PPPoE interfaces not exist anymore, OpenVPN server instances disappeared, etc.
I made a config backup, and noticed the size was only 64 kB instead of about 510 kB normal. I restarted the box remotely but it didn't come back online through internet.
I went to the customer's site to see the box in person with physical console access. There are no IP addresses on public facing interfaces, no default route and was missing config settings for many services.
I went to System/Configuration/History page and reverted config to the version at my colleague's last night last successful edit. Rebooted the box and everything came up working, every service has all configrurations, too.

**To Reproduce**

We didn't do anything that can produce this behavior. The system logs didn't show anything that helps finding the cause of the configuration lost. The logs didn't show any configuration change items between last successful edit (last night) and reboot in the morning. Config history even didn't show changes in the meantime.

**Expected behavior**

Configuration stay unchanged without user interaction.

**Relevant log files**

system.log last lines from 2022-09-26 (last line represents last successful modification. After that time nothing logged)
```
<13>1 2022-09-26T22:16:34+02:00 firewall.pmh.salgotarjan.hu configctl 17462 - [meta sequenceId="96"] event @ 1664223393.58 msg: Sep 26 22:16:33 firewall.pmh.salgotarjan.hu config[4201]: [2022-09-26T22:16:33+02:00][INFO] config-event: new_config /conf/backup/config-1664223393.4956.xml 
<13>1 2022-09-26T22:16:34+02:00 firewall.pmh.salgotarjan.hu configctl 17462 - [meta sequenceId="97"] event @ 1664223393.58 msg: Sep 26 22:16:33 firewall.pmh.salgotarjan.hu config[51517]: [2022-09-26T22:16:33+02:00][INFO] config-event: new_config /conf/backup/config-1664223393.5511.xml 
<13>1 2022-09-26T22:16:34+02:00 firewall.pmh.salgotarjan.hu configctl 17462 - [meta sequenceId="98"] event @ 1664223393.58 exec: system event config_changed
```

configd.log last lines from 2022-09-26:
```
<13>1 2022-09-26T22:16:34+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="90"] [2e66f75f-8a61-44fd-bc69-2b6c3dd85758] trigger config changed event
<13>1 2022-09-26T23:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [adbe377c-38f5-4267-a941-73e3451e48c8] Archive syslog files
<14>1 2022-09-26T23:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message adbe377c-38f5-4267-a941-73e3451e48c8 [syslog.archive] returned OK
```

system.log first lines from 2022-09.27 (reboot issued at 08:10)
```
<13>1 2022-09-27T00:01:59+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="1"] startup, check database.
<13>1 2022-09-27T00:03:35+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="1"] start watching flowd
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="2"] vacuum interface_000030.sqlite
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="3"] vacuum interface_000300.sqlite
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="4"] vacuum interface_003600.sqlite
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="5"] vacuum interface_086400.sqlite
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="6"] vacuum dst_port_000300.sqlite
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="7"] vacuum dst_port_003600.sqlite
<13>1 2022-09-27T00:03:43+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="8"] vacuum dst_port_086400.sqlite
<13>1 2022-09-27T00:03:46+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="9"] vacuum src_addr_details_086400.sqlite
<13>1 2022-09-27T00:04:56+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="1"] vacuum src_addr_000300.sqlite
<13>1 2022-09-27T00:04:56+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="2"] vacuum src_addr_003600.sqlite
<13>1 2022-09-27T00:04:56+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="3"] vacuum src_addr_086400.sqlite
<13>1 2022-09-27T00:05:01+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="4"] vacuum done
<11>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu opnsense 33778 - [meta sequenceId="1"] /usr/local/etc/rc.routing_configure: ROUTING: entering configure using defaults
<13>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu opnsense 33778 - [meta sequenceId="2"] plugins_configure monitor (1)
<13>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu opnsense 33778 - [meta sequenceId="3"] plugins_configure monitor (execute task : dpinger_configure_do(1))
<11>1 2022-09-27T05:01:55+02:00 firewall.pmh.salgotarjan.hu opnsense 75861 - [meta sequenceId="1"] /usr/local/etc/rc.newwanipv6: IPv6 renewal is starting on 'pppoe0'
<11>1 2022-09-27T05:01:55+02:00 firewall.pmh.salgotarjan.hu opnsense 75861 - [meta sequenceId="2"] /usr/local/etc/rc.newwanipv6: Interface '' (pppoe0) is disabled or empty, nothing to do.
<11>1 2022-09-27T05:01:55+02:00 firewall.pmh.salgotarjan.hu opnsense 86855 - [meta sequenceId="3"] /usr/local/etc/rc.newwanip: IPv4 renewal is starting on 'pppoe0'
<11>1 2022-09-27T05:01:55+02:00 firewall.pmh.salgotarjan.hu opnsense 86855 - [meta sequenceId="4"] /usr/local/etc/rc.newwanip: Interface '' (pppoe0) is disabled or empty, nothing to do.
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="1"] vacuum interface_000030.sqlite
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="2"] vacuum interface_000300.sqlite
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="3"] vacuum interface_003600.sqlite
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="4"] vacuum interface_086400.sqlite
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="5"] vacuum dst_port_000300.sqlite
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="6"] vacuum dst_port_003600.sqlite
<13>1 2022-09-27T08:04:42+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="7"] vacuum dst_port_086400.sqlite
<13>1 2022-09-27T08:04:45+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="8"] vacuum src_addr_details_086400.sqlite
<13>1 2022-09-27T08:05:57+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="1"] vacuum src_addr_000300.sqlite
<13>1 2022-09-27T08:05:57+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="2"] vacuum src_addr_003600.sqlite
<13>1 2022-09-27T08:05:58+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="3"] vacuum src_addr_086400.sqlite
<13>1 2022-09-27T08:06:03+02:00 firewall.pmh.salgotarjan.hu flowd_aggregate.py 42321 - [meta sequenceId="4"] vacuum done
<11>1 2022-09-27T08:10:10+02:00 firewall.pmh.salgotarjan.hu opnsense 8936 - [meta sequenceId="1"] /status_services.php: ROUTING: entering configure using defaults
<45>1 2022-09-27T08:19:16+02:00 firewall.pmh.salgotarjan.hu syslog-ng 7981 - [meta sequenceId="1"] syslog-ng shutting down; version='3.37.1'
<45>1 2022-09-27T08:21:58+02:00 firewall.pmh.salgotarjan.hu syslog-ng 15559 - [meta sequenceId="1"] syslog-ng starting up; version='3.37.1'
```
configd.log first lines from 2022-09-27
```
<13>1 2022-09-27T00:00:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [4add9b9c-b7df-4987-b732-34e4df3225f6] cronjob running to sign or renew certificates
<14>1 2022-09-27T00:00:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message 4add9b9c-b7df-4987-b732-34e4df3225f6 [acmeclient.cron-auto-renew] returned OK  
<13>1 2022-09-27T00:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="3"] [5650f7a4-1894-4d1d-a17a-1cf110ca03aa] Archive syslog files
<14>1 2022-09-27T00:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="4"] message 5650f7a4-1894-4d1d-a17a-1cf110ca03aa [syslog.archive] returned OK  
<13>1 2022-09-27T01:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [3dffb599-cd2c-4e3b-9224-f940f9a53bec] Archive syslog files
<14>1 2022-09-27T01:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message 3dffb599-cd2c-4e3b-9224-f940f9a53bec [syslog.archive] returned OK  
<14>1 2022-09-27T01:04:24+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] message 496ea0ef-d0db-4bfe-aa9f-cac47a4d1b5f [firmware.changelog] returned OK  
<13>1 2022-09-27T02:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [d36cae3f-ed04-44e1-9f3c-974eaf481c34] Archive syslog files
<14>1 2022-09-27T02:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message d36cae3f-ed04-44e1-9f3c-974eaf481c34 [syslog.archive] returned OK  
<13>1 2022-09-27T03:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [0dd7d4ab-0280-4846-8431-5f129314c9d6] Archive syslog files
<14>1 2022-09-27T03:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message 0dd7d4ab-0280-4846-8431-5f129314c9d6 [syslog.archive] returned OK  
<13>1 2022-09-27T03:15:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [7c82a12a-dfe8-41d5-989c-f54a801dc34d] Restarting WireGuard
<13>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] [d9a51b39-644f-4308-8062-a690bc67c138] request pf current overall table record count and table-entries limit
<13>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="3"] [8a866e7a-1bd4-4876-b868-65689ed71120] generate template OPNsense/Filter
<13>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="4"] generate template container OPNsense/Filter
<15>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="5"]  OPNsense/Filter generated //usr/local/etc/filter_tables.conf
<15>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="6"]  OPNsense/Filter generated //usr/local/etc/filter_geoip.conf
<13>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="7"] [675fb8f7-7c19-4e6f-bd1b-513cba76e486] refresh url table aliases
<14>1 2022-09-27T03:15:12+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="8"] message 7c82a12a-dfe8-41d5-989c-f54a801dc34d [wireguard.restart] returned OK  
<14>1 2022-09-27T03:15:13+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="9"] message 675fb8f7-7c19-4e6f-bd1b-513cba76e486 [filter.refresh_aliases] returned {"status": "ok"}   
<13>1 2022-09-27T04:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [3a386cff-eefb-45fa-b531-4b310a433b33] Archive syslog files
<14>1 2022-09-27T04:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message 3a386cff-eefb-45fa-b531-4b310a433b33 [syslog.archive] returned OK  
<13>1 2022-09-27T05:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="1"] [927e9779-e2fd-4f24-b6b5-54b4e9ec1895] Archive syslog files
<14>1 2022-09-27T05:01:00+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="2"] message 927e9779-e2fd-4f24-b6b5-54b4e9ec1895 [syslog.archive] returned OK  
<13>1 2022-09-27T05:01:50+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="3"] [34d96d2d-2c01-44ee-b8e3-a48d89187dd1] Rewriting resolv.conf
<13>1 2022-09-27T05:01:51+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="4"] [34d5e6d4-5074-42ee-83fa-a9b57483f51c] Rewriting resolv.conf
<14>1 2022-09-27T05:01:51+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="5"] message 34d96d2d-2c01-44ee-b8e3-a48d89187dd1 [dns.reload] returned OK  
<14>1 2022-09-27T05:01:51+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="6"] message 34d5e6d4-5074-42ee-83fa-a9b57483f51c [dns.reload] returned OK  
<13>1 2022-09-27T05:01:55+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="7"] [076d6ed9-ee9e-4655-bbc8-c710ec4df87d] New IPv6 on pppoe0
<14>1 2022-09-27T05:01:55+02:00 firewall.pmh.salgotarjan.hu configd.py 30135 - [meta sequenceId="8"] message 076d6ed9-ee9e-4655-bbc8-c710ec4df87d [interface.newipv6] returned OK
```

**Additional context**

My Zabbix monitoring server lost communication with this box at 2022-09-27 03:18

**Environment**

HP DL380G9, 1x Xeon E5-2620v3, 16 GB, 2x 500 GB SSD (hardware RAID1), 4x 1 GbE integrated, 2x (2x 1 GBE NIC) add-on cards, all NIC Intel-based, server grade
OPNsense 22.7.1 (amd64, OpenSSL)

I'll give any information required/requested to track this issue further.

**Comment by fichtner** [comment]
There is a traceable history diff under System: Configuration: History you should inspect for why what change occurred. There is little we can do guessing.

**Comment by gallogabor** [comment]
@fichtner 

> There is a traceable history diff under System: Configuration: History you should inspect for why what change occurred. There is little we can do guessing.

Yes, I know that and I looked at it. But there are no relevant change log in that history.

This is where the change happened (in between):

```
9/27/22 08:21:52 | 61 KB | (system): /usr/local/opnsense/mvc/script/run_migrations.php made changes
9/26/22 22:16:33 | 510 KB | ivanhoe@172.16.33.191: /api/freeradius/user/setUser/e69a93f8-4b74-4d3b-8312-49a8f7ef9917 made changes
```
There are no more lines between these two, I copied this from History page unchanged.

At 9/26/22 22:16:33 was the last successful edit and at 9/27/22 08:21:52 was the reboot. But the changes happened hours before the reboot, because Zabbix monitoring (at 03:18) and later the users (at about 07:45) noticed the disfunctioning firewall earlier.

Which log files could help me find out what happened?

I run OPNsense for about 7 years on many-many boxes and never happened anything like this to anyone of them.

**Comment by derbolle** [comment]
A similar issue appeared with our Opnsense 22.7.5 installation yesterday when an acme renewal and cronjob which restarts haproxy was triggered. Seemingly random things were affected e.g. haproxy frontend vanished and some firewall rules changed. Also one of our letsencrypt certificate configurations disappeared. All in all the size of the config file changed from 631kb to 628kb. We will try and revert to an earlier config backup tomorrow night.

**Comment by fraenki** [comment]
To me this sounds very much like a filesystem corruption. If the system has not been rebooted yet, check with `dmesg` on the command line to see if there are any filesystem-related errors or warnings.

If this is a ZFS filesystem, verify filesystem integrity by performing a "scrub":

```
# start scrub
zpool scrub zroot

# wait for scrub to complete, then check pool status
zpool status
```

Note that performing a "scrub" may cause severe performance degration.

**Comment by rudiservo** [comment]
@fraenki Uh... 2 things cross my mind, yes I am on ZFS, why FS corruption? It's on an SSD with no errors and ZFS is COW, I haved used and abused BTRFS with no issues at all.

Anyway, there is a cron option for scrub, left it running every night.
